src/Microsoft.PowerShell.Commands.Management/commands/management/ComputerUnix.cs

```diff
@@ -0,0 +1,164 @@
+using Dbg = System.Management.Automation;
+
+namespace Microsoft.PowerShell.Commands
+{
+    #region Stop-Computer
+
+    /// <summary>
+    /// Cmdlet to stop computer.
+    /// </summary>
+    [Cmdlet(VerbsLifecycle.Stop, "Computer", SupportsShouldProcess = true,
+        HelpUri = "https://go.microsoft.com/fwlink/?LinkID=135263", RemotingCapability = RemotingCapability.SupportedByCommand)]
+    public sealed class StopComputerCommand : PSCmdlet, IDisposable
+    {
+        #region Private Members
+
+        private readonly CancellationTokenSource _cancel = new CancellationTokenSource();
+        private const int forcedShutdown = 5; // See https://msdn.microsoft.com/library/aa394058(v=vs.85).aspx
+
+        #endregion
+
+        #region "Parameters"
+
+        /// <summary>
+        /// The authentication options for CIM_WSMan connection.
+        /// </summary>
+        [Parameter]
+        [ValidateSet(
+            "Default",
+            "Basic",
+            "Negotiate", // can be used with and without credential (without -> PSRP mapped to NegotiateWithImplicitCredential)
+            "CredSSP",
+            "Digest",
+            "Kerberos")] // can be used with explicit or implicit credential
+        [SuppressMessage("Microsoft.Naming", "CA1704:IdentifiersShouldBeSpelledCorrectly")]
+        public string WsmanAuthentication { get; set; } = "Default";
+
+        /// <summary>
+        /// The following is the definition of the input parameter "ComputerName".
+        /// Value of the address requested. The form of the value can be either the
+        /// computer name ("wxyz1234"), IPv4 address ("192.168.177.124"), or IPv6
+        /// address ("2010:836B:4179::836B:4179").
+        /// </summary>
+        [Parameter(Position = 0, ValueFromPipelineByPropertyName = true)]
+        [ValidateNotNullOrEmpty]
+        [SuppressMessage("Microsoft.Performance", "CA1819:PropertiesShouldNotReturnArrays")]
+        [Alias("CN", "__SERVER", "Server", "IPAddress")]
+        public string[] ComputerName { get; set; } = new string[] { "." };
+
+        /// <summary>
+        /// The following is the definition of the input parameter "Credential".
+        /// Specifies a user account that has permission to perform this action. Type a
+        /// user-name, such as "User01" or "Domain01\User01", or enter a PSCredential
+        /// object, such as one from the Get-Credential cmdlet.
+        /// </summary>
+        [Parameter(Position = 1)]
+        [ValidateNotNullOrEmpty]
+        [Credential]
+        public PSCredential Credential { get; set; }
+
+        /// <summary>
+        /// Force the operation to take place if possible.
+        /// </summary>
+        [Parameter]
+        public SwitchParameter Force { get; set; } = false;
+
+        #endregion "parameters"
+
+        #region "IDisposable Members"
+
+        /// <summary>
+        /// Dispose Method.
+        /// </summary>
+        public void Dispose()
+        {
+            try
```

Comment:
I don't think try/catch is necessary here.  Dispose() can be called multiple times (I believe `ObjectDisposedException` is only thrown when a disposed object is used).

Reply:
This is copy/paste from `Computer.cs` so if that's true it should be fixed in both places, I'm not enough of a C# expert to know which is right.

Reply:
The documentation I found says Dispose() 'typically' does not throw the `ObjectDisposedException`.  I seem to remember other documentation saying Dispose() should not throw.  But most (I guess not all :)) of our code base assumes it does not throw and I think we should follow that pattern here.

Reply:
I fixed it here (and in `Computer.cs`), please take another look.